core/templates/pages/contributor-dashboard-page/services/translate-text.service.ts

```diff
@@ -50,36 +52,42 @@ export class StateAndContent {
     public translation: string | string[],
     public dataFormat: string,
     public contentType: string,
-    public interactionId?: string,
-    public ruleType?: string
+    // The following two properties are set to null when the
+    // interactionId and ruleType is not provided.
```

Comment:
Why they are not provided?

Reply:
"translatableItem.interactionId, translatableItem.ruleType " is assigned to a type string | null, so to remove the conflict I added null as a type to them.
This was the main reason for assigning null as a type to them.

Reply:
Okay, but you should explain in what case they can be `null`.